packages/platform-browser/src/dom/dom_renderer.ts

```diff
@@ -206,7 +209,9 @@ class DefaultDomRenderer2 implements Renderer2 {
     let el: any = typeof selectorOrNode === 'string' ? document.querySelector(selectorOrNode) :
                                                        selectorOrNode;
     if (!el) {
-      throw new Error(`The selector "${selectorOrNode}" did not match any elements`);
+      throw new RuntimeError(
+          RuntimeErrorCode.ROOT_SELECTOR_NO_MATCH,
+          ngDevMode && `The selector "${selectorOrNode}" did not match any elements`);
```

Comment:
Looks great and let's also use `NG_DEV_MODE` as mentioned above.
```suggestion
          NG_DEV_MODE && `The selector "${selectorOrNode}" did not match any elements`);
```